hudi-client/hudi-java-client/src/test/java/org/apache/hudi/hadoop/TestHoodieFileGroupReaderOnHive.java

```diff
@@ -104,11 +106,13 @@ public void assertRecordsEqual(Schema schema, ArrayWritable expected, ArrayWrita
     ArrayWritableTestUtil.assertArrayWritableEqual(schema, expected, actual, false);
   }
 
-  private void setupJobconf(JobConf jobConf) {
-    Schema schema = HoodieAvroUtils.addMetadataFields(HoodieTestDataGenerator.AVRO_SCHEMA);
+  private void setupJobconf(JobConf jobConf, Schema schema) {
     List<Schema.Field> fields = schema.getFields();
     setHiveColumnNameProps(fields, jobConf, USE_FAKE_PARTITION);
-    jobConf.set("columns.types","string,string,string,string,string," + HoodieTestDataGenerator.TRIP_HIVE_COLUMN_TYPES + ",string");
+    List<TypeInfo> types = TypeInfoUtils.getTypeInfosFromTypeString(HoodieTestDataGenerator.TRIP_HIVE_COLUMN_TYPES);
+    Map<String, String> typeMappings = HoodieTestDataGenerator.AVRO_SCHEMA.getFields().stream().collect(Collectors.toMap(Schema.Field::name, field -> types.get(field.pos()).getTypeName()));
+    String columnTypes = fields.stream().map(field -> typeMappings.getOrDefault(field.name(), "string")).collect(Collectors.joining(","));
+    jobConf.set("columns.types", columnTypes + ",string");
```

Comment:
Do these configs need to match the columns in the query?  Could we still test the Hive queries with meta columns in some way?

Reply:
The schema provided will have the meta fields when `populateMetaFields` is true